### PR TITLE
Ensure things that should be ints are ints

### PR DESCRIPTION
**Pull request by f0uriest**
- Adds checking in a bunch of places to ensure stuff like L,M,N,NFP are ints
- Adds int casting to `_set_up` methods for backwards compatibility
- Removes some setter methods, as they're redundant with `change_resolution` and weren't implemented uniformly across classes/attributes.

### CODE REVIEW

**Review by ddudt — APPROVED** [review]

File: desc/basis.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 from desc.backend import cond, custom_jvp, fori_loop, gammaln, jit, jnp, sign, switch
 from desc.io import IOAble
-from desc.utils import flatten_list
+from desc.utils import check_nonnegint, check_posint, flatten_list
```
Comment (rahulgaur104) [inline desc/basis.py:12]: Wasn't there something like isposint somewhere? Why not use that?

File: desc/basis.py [hunk 2]
```diff
@@ -252,9 +233,9 @@ class PowerSeries(_Basis):
     """
 
     def __init__(self, L, sym="even"):
-        self.L = L
-        self.M = 0
-        self.N = 0
+        self._L = check_nonnegint(L, "L", False)
+        self._M = 0
```
Comment (rahulgaur104) [inline desc/basis.py:237]: Why no check_nonnegint for _M and _N?

File: desc/basis.py [hunk 3]
```diff
@@ -647,9 +630,9 @@ class ZernikePolynomial(_Basis):
     """
 
     def __init__(self, L, M, sym=False, spectral_indexing="ansi"):
-        self.L = L
-        self.M = M
-        self.N = 0
+        self._L = check_nonnegint(L, "L", False)
+        self._M = check_nonnegint(M, "M", False)
+        self._N = 0
```
Comment (rahulgaur104) [inline desc/basis.py:635]: Why no check here?

File: tests/test_transform.py [hunk 4]
```diff
@@ -197,9 +197,9 @@ def test_set_basis(self):
         """Tests the basis setter method."""
         grid = ConcentricGrid(L=4, M=2, N=1)
 
-        basis_20 = FourierZernikeBasis(L=-1, M=2, N=0)
```
Comment (dpanici) [inline tests/test_transform.py:200]: do you know why these were -1 before?

**Review by dpanici — APPROVED** [review]

File: tests/test_transform.py (hunk 4 above)
Comment (dpanici) [reply tests/test_transform.py:200]: oh was it to set it to the default based on the indexing?

File: desc/basis.py (hunk 2 above)
Comment (f0uriest) [reply desc/basis.py:237]: they're hardcoded to 0 for this class

File: desc/basis.py (hunk 3 above)
Comment (f0uriest) [reply desc/basis.py:635]: see above

File: tests/test_transform.py (hunk 4 above)
Comment (f0uriest) [reply tests/test_transform.py:200]: yeah `L=-1` meant `M if spectral_indexing=="ansi" else 2*M` but it wasn't really used anywhere and its better to be explicit.

**Pull request merged**